crates/tx5-connection/src/webrtc/go_pion.rs

```diff
@@ -124,8 +147,39 @@ async fn task(
             use tx5_go_pion::PeerConnectionEvent as Evt;
             while let Some(evt) = peer_evt.recv().await {
                 match evt {
-                    Evt::Error(_) => break,
-                    Evt::State(_) => (),
+                    Evt::Error(err) => {
+                        cmd_send2.send_or_close(Cmd::Error(
+                            std::io::Error::other(format!(
+                                "DataChanError: {err}"
+                            ))
+                        ))?;
+                    },
+                    Evt::State(state) => {
+                        match state {
+                            tx5_go_pion::PeerConnectionState::Disconnected => {
+                                cmd_send2.send_or_close(Cmd::Error(
+                                    std::io::Error::other(
+                                        "PeerConnectionState changed to Disconnected",
+                                    ),
+                                ))?;
+                            },
+                            tx5_go_pion::PeerConnectionState::Closed => {
+                                cmd_send2.send_or_close(Cmd::Error(
+                                    std::io::Error::other(
+                                        "PeerConnectionState changed to Closed",
+                                    ),
+                                ))?;
+                            },
```

Comment:
I question these being handled as errors. I see that this matches the behavior in the data channel implementation but it will lead to a `tracing::error!` with a "WebRTC task failed" message. That seems misleading to me. Should these be their own enum variant to communicate stopping without an error?

Possibly something that should be considered in the data channel implementation too.

Reply:
That makes sense to me that this is proper handling of expected behavior rather than an error.

I would suggest doing that refactoring in a separate PR because I'm guessing it will be a wider scope. I believe we're relying on the errors propagating up to trigger dropping the connection, but I haven't combed through the codebase to see it clearly.

Reply:
The errors are staying in this file and just being logged. What we rely on is the drop behavior. I don't think the scope of that change would be any more significant than what you've already done by adding the `Error` variant. On another PR is fine but then can we track it please :)

Reply:
I'm can't figure out how to do a proper integration test that exercises this. Open to any suggestions.